src/main/scala/zio/prelude/Derive.scala

```diff
@@ -56,6 +59,15 @@ object Derive {
         Equal.EitherEqual
     }
 
+  /**
+   * The `DeriveEqual` instance for `Future`.
+   */
+  implicit val FutureDeriveEqual: DeriveEqual[Future] =
```

Comment:
I think I would leave out the `Equal` instance for `Future` here. I understand wanting to test the laws for it but defining an `Equal` instance implies there is a sensible way to compare `Future` values for equality. If all `Future` did was asynchronously return a pure value that could make sense. But I think `Future` is normally used as a quasi-effect type where within `Future` you could be doing anything so the fact that two `Future` values return the same result doesn't mean they are the same (maybe one also launched a web query and the other also send some information to a database). For the same reason we haven't defined an `Equal` instance for ZIO and I think the fact that you are having to call `unsafeRun` here reflects the same problem. We have had some discussion of having a concept of `TestEqual` purely for testing but I think we want to keep that separate from actual equality.

Reply:
Yeah, makes sense, I'll just remove this for now